src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -93,18 +93,56 @@ def details_widget
                 _("Select an extension or a module to show details here") + "</small>")))
       end
 
+      # @return [String] a Value for a RichText
+      def richtext_checkboxes(addons)
+        items = addons.map do |addon|
+          # checkbox label for an unavailable extension
+          # (%s is an extension name)
+          label = addon.available? ? addon.label : (_("%s (not available)") % addon.label)
+
+          rt_cb(id: addon_widget_id(addon),
+                label: label,
+                selected: addon_selected?(addon),
+                enabled: addon.available?,
+                indented: addon.depends_on)
+        end
+        items.join("\n")
+      end
+
+      # FIXME acknowledge Lada's code
+      # https://gist.github.com/lslezak/5ed82fe19337bef4807b
+
+      IMG_ON = "/usr/share/YaST2/theme/current/wizard/checkbox-on.png"
+      IMG_OFF = "/usr/share/YaST2/theme/current/wizard/checkbox-off.png"
+
+      def rt_cb(id:, label:, selected:, enabled:,
+                indented: false, text_mode: Yast::UI.TextMode)
+        selected = false unless enabled
+        if text_mode
+          indent = "&nbsp;" * (indented ? 5 : 1)
+          check = selected ? "[x]" : "[ ]"
+          widget = "#{check} #{label}"
+          enabled_widget = enabled ? "<a href=\"#{id}\">#{widget}</a>" : widget
+        else
+          indent = "&nbsp;" * (indented ? 7 : 1)
+          check = "<img src='#{selected ? IMG_ON : IMG_OFF}'></img>"
+          widget = "#{check} #{label}"
+          enabled_widget = if enabled
+            "<a href='#{id}' style='text-decoration:none; color:black'>#{widget}</a>"
+          else
+            "<span style='color:grey'>#{widget}</span>"
+          end
+        end
+        "<p>#{indent}#{enabled_widget}</p>"
+      end
+
       # create UI box with addon check boxes, if the number of the addons is too big
       # the UI uses two column layout
       # @return [Yast::Term] the main UI dialog term
       def addons_box
-        lines = Yast::UI.TextMode ? 9 : 14
-        if @addons.size <= lines
-          content = addon_selection_items(@addons)
-        else
-          content = two_column_layout(@addons[lines..(2 * lines - 1)], @addons[0..(lines - 1)])
-        end
+        content = RichText(Id(:items), richtext_checkboxes(@addons))
```

Comment:
when you modify UI, can you please also add to top link to enable/disable beta add-ons filtering? I think it fit there and to avoid conflicts I do not add it to my branch.